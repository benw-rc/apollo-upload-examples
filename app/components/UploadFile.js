import gql from 'graphql-tag'
import { graphql } from 'react-apollo'

const S3_DOCUMENT_STORE_BUCKET = 'my-bucket'
const S3_DOCUMENT_STORE_KEY = 'reqs'

let s3Location = {}
s3Location = {
  bucket: S3_DOCUMENT_STORE_BUCKET,
  key: S3_DOCUMENT_STORE_KEY
}

const UPLOAD_FILE = gql`
  mutation singleUpload($input: DocumentInput!) {
    singleUpload(input: $input) {
      id
      fileName
      uploadedBy
      uploadedAt
    }
  }
`

const uploadOneFile = ({ mutate }) => {
  const handleChange = ({
    target: {
      validity,
      files: [file]
    }
  }) => {
    //eslint-disable-next-line
    console.log('iupload: ', file)
    const input = {
      config: {
        AMPLICON: 'amplicon',
        EMAILS: 'al@rc.com'
      },
      file,
      s3Location
    }

    return (
      validity.valid &&
      mutate({
        variables: { input }
      })
    )
  }

  return <input type="file" required onChange={handleChange} />
}

const enhancedUploadOneFile = graphql(UPLOAD_FILE)
const uploadFile = enhancedUploadOneFile(uploadOneFile)

export default uploadFile
