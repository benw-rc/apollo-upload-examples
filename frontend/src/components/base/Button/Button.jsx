/** @module */
import React from 'react';
import { Button, Icon } from 'semantic-ui-react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import Component from '../Component';

import './Button.scss';

/**
 * Basic Button implementation.  Adds our styling options to the basic
 * semantic-ui-react Button.
 *
 * Accepts all options for the semantic-ui-button:
 * <https://react.semantic-ui.com/elements/button>
 *
 * @param {boolean} [active]  - is the button active? (adds css class)
 * @param {string} [className] - additional css class names
 * @param {string} [icon] - icon name
 * @param {boolean} [inverted=false]- (optional [ default: false ])
 */
export class RCButton extends Component {
  render() {
    const { className, children, active, icon, ...props } = this.props;
    const btn_props = {
      inverted: true,
      size: 'mini',
      className: classNames(className, 'argus-btn', { active }),
      ...props,
    };
    if (children === undefined) {
      return <Button icon={icon} { ...btn_props } />
    }
    return (
      <Button { ...btn_props } >
        { icon !== undefined && <Icon name={icon} /> }
        { children }
      </Button>
    );
  }
}
RCButton.propTypes = {
  className: PropTypes.string,
  active: PropTypes.bool,
  icon: PropTypes.string,
}

export default RCButton;
