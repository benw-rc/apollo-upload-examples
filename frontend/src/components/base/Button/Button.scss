@import "../../../scss/variables.scss";
.ui.button.argus-btn {
  box-shadow: 0 0 0 1px #fff inset !important;
  color: #fff;
  text-shadow: none !important;
  &:focus {
    box-shadow: 0 0 0 1px #fff inset !important;
    text-shadow: inherit !important;
    &.active {
      color: black !important;
    }
    &.inverted{
      box-shadow: none;
    }
    &.inverted:not(:active):not(.active) {
      &.blue {
        box-shadow: 0px 0px 0px 2px $blue inset !important;
      }
      &.red {
        box-shadow: 0px 0px 0px 2px $red inset !important;
      }
      &.yellow {
        box-shadow: 0px 0px 0px 2px $yellow inset !important;
      }
      &.green {
        box-shadow: 0px 0px 0px 2px $green inset !important;
      }
      &.orange {
        box-shadow: 0px 0px 0px 2px $orange inset !important;
      }
      &:not(.blue):not(.red):not(.yellow):not(.green):not(.orange) {
        color: white !important;
      }
      background-color: transparent !important;
    }
  }
  &.active {
    background-color: $panel-v-light !important;
  }

}
