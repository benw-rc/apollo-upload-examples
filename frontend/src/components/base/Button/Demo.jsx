import React from 'react';
import { ButtonGroup } from 'semantic-ui-react';

import Component from '../Component';

import Button from './Button';

export class Demo extends Component {
  state = {
    toggle_val: false,
  }
  toggle_val = () => this.setState({ toggle_val: !this.state.toggle_val });
  render() {
    return (
      <div>
        <h4>Styling</h4>
        <Button>Simple Dummy Button</Button>
        <Button color="blue">Colored Button</Button>
        <Button color="blue" inverted={false}>Bold button</Button>
        <Button icon="cogs">Icon Button</Button>
        <Button icon="expand" />
        <Button active>Active button</Button>
        <h4>Button Group</h4>
        <ButtonGroup>
          <Button icon="expand" />
          <Button icon="map">Icon Button</Button>
          <Button color="green">Colored Button</Button>
        </ButtonGroup>
        <h4>Connected: Toggle value = {this.state.toggle_val ? 'True' : 'False'}</h4>
        <Button onClick={this.toggle_val}>Toggle Value</Button>
      </div>
    );
  }
}

export default Demo;
