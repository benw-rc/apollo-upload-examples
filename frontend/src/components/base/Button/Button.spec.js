import React from 'react';
import { render, simple_test } from 'rc-modules/redux/test/components';

import { RCButton as Button } from './Button';

jest.mock('semantic-ui-react');

describe('Button component', () => {
  let el, sub_el, el_props;
  const props = { some: 'component', props: 'for testing' };
  const className = 'cl_n';
  const icon = 'Icon';
  const children = <div>Some Stuff</div>

  simple_test('basic', Button, { className, icon, ...props })
  simple_test('with_children', Button, { className, icon, children, ...props })

  describe('className', () => {
    it('uses passed className alongside argus-btn className', () => {
      el = render(Button, { className });
      expect(el.hasClass('argus-btn')).toEqual(true);
      expect(el.hasClass(className)).toEqual(true);
    });
    it('includes active className iff that props is included', () => {
      el = render(Button, { className });
      expect(el.hasClass('active')).toEqual(false);
      el = render(Button, { className, active: true });
      expect(el.hasClass('active')).toEqual(true);
    });
  });

  describe('icon and children', () => {
    it('passes icon as a prop if no children are passed', () => {
      el = render(Button, { className, icon });
      expect(el.prop('icon')).toEqual(icon);
      expect(el.children().exists()).toEqual(false)
    });
    it('passes icon as a child if other children are passed', () => {
      el = render(Button, { className, icon, children });
      expect(el.prop('icon')).toEqual(undefined);
      expect(el.children().contains(children));
      const icon_el = el.children().find('Icon');
      expect(icon_el.exists()).toEqual(true);
      expect(icon_el.prop('name')).toEqual(icon);
    });
  });
});
