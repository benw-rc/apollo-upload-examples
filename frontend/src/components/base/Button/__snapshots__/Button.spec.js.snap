// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component :basic 1`] = `
<Button
  as="button"
  className="cl_n argus-btn"
  icon="Icon"
  inverted={true}
  props="for testing"
  size="mini"
  some="component"
/>
`;

exports[`Button component :with_children 1`] = `
<Button
  as="button"
  className="cl_n argus-btn"
  inverted={true}
  props="for testing"
  size="mini"
  some="component"
>
  <Icon
    as="i"
    name="Icon"
  />
  <div>
    Some Stuff
  </div>
</Button>
`;
